### PR TITLE
Enhancements and Testing Summary

### PR DESCRIPTION
**Pull request by lily4178993**
# Enhancements and Testing Summary

## Overview

In the **Space Travelers' Hub** project, our collaborative team consisting of @mryeminthu  and @lily4178993 has successfully developed a web application that empowers users to book rockets and participate in space missions.

## Components and Features

Our project revolves around three main sections: **Rockets**, **Missions**, and **My Profile**.

### Rockets

The Rockets section presents users with a curated list of available SpaceX rockets, offering the capability to reserve or cancel reservations. The user interface showcases essential rocket details, such as names, descriptions, and captivating images.

Web capture Rockets

### Missions

The Missions section provides users with a glimpse into the exciting world of space exploration. Users can explore current space missions, read engaging descriptions, and seamlessly join or leave missions, enhancing their immersive experience.

Web capture Missions

### My Profile

The My Profile section tailors the user experience by offering a personalized overview of reserved rockets and joined missions. This intuitive interface empowers users to effortlessly track their participation history and journey into the cosmos.

Web capture Profile

## Implementation Highlights

Our team has demonstrated a dedicated focus on implementing key functionalities that ensure a seamless and captivating user experience:

- Real-time data retrieval from the SpaceX API for Rockets, Missions.
- Strategic utilization of Redux and Redux Toolkit to effectively manage and store fetched data, ensuring streamlined state management.
- Clever deployment of conditional rendering, dynamically displaying reserved and joined statuses for rockets, and missions.
- Seamless navigation facilitated by the integration of React Router, offering users an engaging and dynamic journey.

## Testing Efforts

Our commitment to delivering a reliable and stable application led us to invest substantial effort into testing. We conducted thorough unit and integration tests for critical sections, including navigation(**Header**), **MyProfile**, **Missions** and **Rockets**. This rigorous testing process encompassed simulating user interactions, validating data integrity, and ensuring smooth component rendering.

## Notes for Reviewers

Reviewers are encouraged to explore the application firsthand using the deployed [link](https://nelye-spacetravellershub.netlify.app/) or by setting up a local environment with the following commands:

```bash
git clone https://github.com/lily4178993/space-travellers.git
cd space-travellers
npm install
npx eslint "**/*.{js,jsx}" --fix
npm start
```

### CODE REVIEW

File: src/components/Rockets.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const dispatch = useDispatch();
+  const rocketData = useSelector((state) => state.rockets);
+
+  useEffect(() => {
+    if (rocketData.length === 0) {
+      fetch('https://api.spacexdata.com/v3/rockets')
+        .then((response) => response.json())
+        .then((data) => {
+          dispatch(fetchRockets(data));
+        })
```
Comment (DammyShittu) [inline src/components/Rockets.jsx]: - [ ] Kindly filter the rocket data from the API before dispatching it to the store, based on [this requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store).

  image

File: src/redux/missions/missionsSlice.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+const MISSION_API_URL = 'https://api.spacexdata.com/v3/missions';
+
+/**
+ * An asynchronous Redux thunk action creator to fetch missions from the SpaceX API.
+ *
+ * @type {AsyncThunk<Array, void, {}>}
+ */
+const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
+  const response = await axios.get(MISSION_API_URL);
+  return response.data;
```
Comment (DammyShittu) [inline src/redux/missions/missionsSlice.js]: - [ ] Kindly filter the data from the missions API too. It is not all the information from the API that should be dispatched to the store.

  image

File: package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "devDependencies": {
+    "@babel/core": "^7.23.0",
+    "@babel/eslint-parser": "^7.22.15",
+    "@babel/plugin-syntax-jsx": "^7.22.5",
+    "@babel/preset-react": "^7.22.15",
+    "eslint": "^7.32.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-jsx-a11y": "^6.7.1",
```
Comment (DammyShittu) [inline package.json]: - [ ] From the requirement, the redux logger should be present in the `package.json` file. Kindly add it. When you add it and configure the store with it, you can view the store in the browser console.

  ```
  npm install -S redux-logger
  ```

File: src/components/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  <header>
+    <div className="logo-container">
+      <img src={logo} alt="Space Travelers' Hub Logo" />
+      <h1>Space Travelers&apos; Hub</h1>
+    </div>
+    <nav>
+      <ul>
+        <li><Link to="/">Rockets</Link></li>
+        <li><Link to="/missions">Missions</Link></li>
+        <li><Link to="/profile">My Profile</Link></li>
```
Comment (DammyShittu) [inline src/components/Header.jsx]: - [ ] From the requirement, you should use <NavLink>, not <Link> for your header links. NavLink has the benefit of allowing us style the selected menu option, which is also a requirement.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @lily4178993 and @mryeminthu,

## Highlights
✔️ API is fetched once

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lily4178993** [comment]
Thank you @DammyShittu for your thorough inspection and valuable suggestions. Your keen eye for detail and commitment to quality have greatly contributed to the improvement of this project. We appreciate your dedication to excellence!

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Detailed pull request message ✔️
- The application work as expected ✔️
- No liner errors ✔️
apprrrr

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by lily4178993** [comment]
We want to express our sincere gratitude to the reviewers for their careful evaluation and approval of this project. Your feedback and guidance have been invaluable in ensuring the success of this endeavor. Thank you for your time, insights, and support throughout this journey!

**Pull request merged**